Content.Shared/Pinpointer/SharedNavMapSystem.cs

```diff
@@ -81,6 +90,38 @@ protected bool TryCreateNavMapBeaconData(EntityUid uid, NavMapBeaconComponent co
         return true;
     }
 
+    public void RequestWarpTo(EntityUid uid, Vector2 target)
+    {
+        var message = new MapWarpRequest(GetNetEntity(uid), target);
+        RaiseNetworkEvent(message);
+    }
+
+    private void OnMapWarp(MapWarpRequest args)
+    {
+        var uid = GetEntity(args.Uid);
+
+        // This was only tested with the AI Eye but should theoretically work with any other future remote-controlled things
+        if (TryComp<EyeComponent>(uid, out var eye) && eye.Target is not null)
+            uid = eye.Target.Value;
+
+        if (!TryComp<NavMapWarpComponent>(uid, out var warpComp))
+            return;
+
+        var xform = Transform(uid);
+
+        if (xform.MapUid is null)
+            return;
+
+        var index = _random.Next(warpComp.Sounds.Count);
+        _audio.PlayGlobal(warpComp.Sounds[index],
+            GetEntity(args.Uid),
+            AudioParams.Default.WithVariation(warpComp.PitchVariation));
+
+        // This was designed for incorporeal entities, thus there aren't any collision checks or anything
+        _transformSystem.SetCoordinates(uid, xform, new EntityCoordinates(xform.MapUid.Value, args.Target));
+        _transformSystem.AttachToGridOrMap(uid, xform);
+    }
+
```

Comment:
I think you're better off just making it a BUI event so it gets validation done.

Also the audio should be predicted instead. Also don't do multiple transformsystem calls as it will double the amount of moveevents that get raised which are extremely expensive, just set the position once. Use PlaceAt or the likes.

Reply:
Audio is now predicted. I have removed the AttachToGridOrMap call entirely, because it does not actually seem to bee needed at all (it was mostly just skipped by checks anyway). 

There is now also a 2 second "cooldown" on warps so they can't be spammed